### ISSUE TITLE
Unnecessary Trash Convention Principles

### ISSUE THREAD

**Issue by Dr-Kakashi** [Zamiell/hanabi-conventions#237]
**Unnecessary Trash Conventions Principles** 
by Dr_Kakashi and Florian

Florian and I have been working on this and it's time to just standardize everything and add some trash principles, so that it flows without having to memorize conventions. Maybe even simplify the doc a bit and remove the long-named conventions related to this and just give it it's own section.  Basically, only the foundational conventions should be listed.  ie. trash push, trash bluff, etc.  No need to list all the different types of unnecessary trash bluffs, for example.

There are two types of “situations” for when a clue is "unnecessary."
1.	Cluing trash that is not yet known.  Thus, as a side note, it is unnecessary from the perspective of the person who receives the clue (Unnecessary A). This is more for things where the next person already has a blind play and this blind play could've been obtained differently (a simple play clue).
a.	Trash Bluff
b.	UTD
c.     UTPD
d.     etc.

2.	Cluing trash that is known trash.  So, unnecessary for the rest of the team (Unnecessary B). This is for when the card of the clue receiver could've been gotten differently (a simple play clue). 
a.	Trash chop move (on a card which can be saved directly)
b.	Trash Push (which converts into unnecessary trash push)
c.     Junk Discharge
d.	Etc.

**Changes:**
In the situation of UA, the clue receiver should convert it to the known trash variant of the clue.  
1.	Unnecessary trash bluff or UTD should become a chop move/trash push depending on the location of the trash
2.	~~If this interpretation causes the clue receiver to chop move trash/bomb, then someone else should blind play slot 1 to “undo” the trash convention.
a. For example a trash bluff is initiated by Alice. Bob blind plays slot 1. If Cathy does nothing, Donald will chop move trash. Cathy blind plays slot 1. Donald realizes that this is the double trash bluff (form 1) and undo's the chop move.~~

In the situation of UB, if a normal play/save clue could’ve been given but using a trash convention to chop move or trash push, then it calls for an extra blind play.
1.	Using known trash to cm a 5 and some critical card.  In this situation, 5cm could’ve been given.  In this case, a blind play is required from the immediate next player. (Unnecessary Trash Chop Move)
2.	Using known trash to trash push a card that could’ve been given a normal play clue.  In this case, a blind play is required from the immediate next player. (Unnecessary Trash Push)
3.	Another example: an Unnecessary Trash Push Discharge is initiated on Cathy by Alice.  Bob successfully discharges.  Cathy can see that the card that Bob played could've easily been clued.  She understands that it's super strange for Alice to do such a complicated move on a card that can be simply play clued.  Cathy blind play slot 1.  

Edit: 
All trash conventions should follow Occam's Razor
https://github.com/Zamiell/hanabi-conventions/blob/master/Reference.md#clue-interpretation--occams-razor
- Basically, we should always assume a finesse over a bluff.  
- Also, the situation that gets the least amount of blind cards played shall always take precedence

Example:
https://github.com/Zamiell/hanabi-conventions/issues/220

Edit 2:
These principles are mainly for early-mid game.  Shall we make an exception where, once end game begins, only the trash double bluff conventions shall be used and unnecessary principles turns off?

**Comment by Dr-Kakashi** [comment]
reserve

**Comment by Dr-Kakashi** [comment]
As an added benefit....we could potentially just delete the long-named conventions from the doc and just have this in it's place.  

The Unnecessary Unknown Trash Discharge Chop Move (UUTDCM), The Unnecessary Trash Finesse Chop Move & The Unnecessary Trash Bluff Chop Move, etc. for example.

**Comment by Dr-Kakashi** [comment]
![2020-05-26_0811](https://user-images.githubusercontent.com/48993588/82917631-85c4d900-9f28-11ea-9174-9389ae7e3746.png)
![2020-05-26_0811_001](https://user-images.githubusercontent.com/48993588/82917640-878e9c80-9f28-11ea-9218-046be31fd092.png)

So I'm showing these two situations to show the issue I'm seeing with how trash clues are being interpreted.  

I've made it so that everyone has a play on finesse position and kimbifille has a playable card next to the trash on chop.

So the first image kimbifille is given "known trash" on chop.  This is interpreted as the Unnecessary Trash Push Convention

The second image kimbifille is given "unknown trash" on chop.  This is interpreted as the Trash Double Bluff (form 1) Convention.

**Comment by iamwhoiamhahaha** [comment]
Seems fine. And I believe it's time to generalize all the trash conventions. Good job, kakashi and florian! Let's see how zamiel would respond.

**Comment by Zamiell** [comment]
sounds like you should close #235 and #236

**Comment by Zamiell** [comment]
and #220

**Comment by Zamiell** [comment]
and #233

**Comment by Zamiell** [comment]
correct me if i am wrong

**Comment by pianoblook** [comment]
Excited to learn about all this more. I think a relevant situation came up in my game today ( from https://hanabi.live/replay/182667/45 ) which I wanted to share: in this case a delayed (Type UB) Unnecessary Trash Push.

1's to ADrone looks like it should get read as a UTPF, but Huxley delays the play by playing g4 to set up g5. Something weird like this is the only way to get both cards played in time. (Zamiel pointed out that an Unnecessary Trash Pull w/ '2' also would work, but this feels like a similarly weird situation)

<img width="908" alt="Screen Shot 2020-05-26 at 4 21 17 PM" src="https://user-images.githubusercontent.com/32303453/82948220-6604d380-9f6f-11ea-8847-36e7a78c89dc.png">

**Comment by Dr-Kakashi** [comment]
Yes, that situation only works because Adrone has no time to clue the B5 directly.

**Comment by florian-5f3759df** [comment]
Here's a summary of what I think the interaction with the current conventions is, if this is implemented. I didn't observe clear conflicts with the current set (except with some trash double bluffs in endgame) so far.

_Unnecessarily Using Unknown Trash_ (UA in the proposal, the best name has still be figured out) is a set of rules which replaces / generalizes:
* [*Unnecessary Trash Finesse Chop Move & The Unnecessary Trash Bluff Chop Move*](https://github.com/Zamiell/hanabi-conventions/blob/master/Reference.md#the-unnecessary-trash-finesse-chop-move--the-unnecessary-trash-bluff-chop-move)
* [*Unnecessary Unknown Trash Discharge Chop Move (UUTDCM)*](https://github.com/Zamiell/hanabi-conventions/blob/master/Reference.md#the-unnecessary-unknown-trash-discharge-chop-move-uutdcm)
* [*Trash Double Bluff*](https://github.com/Zamiell/hanabi-conventions/blob/master/Reference.md#the-trash-double-bluff-form-1
) (although some clues that were double bluff now default to a different interpretation)

Adds: (Without having to be explicitly named, these can just be performed from the general rule.)
* *Unnecessary Trash Bluff Trash Push* #235 
* *Unnecessary UTD Trash Push* #236 
* *Unnecessary UTD Bluff* (if the trash interpretation chop moves trash / trash pushes an unplayable card, the next player blind plays, so this is a double bluff which makes Bob eject and gets an additional blind play by Cathy)

In rare cases, it can get three cards in 5 or 6 player (by Alice doing an "Unnecessary Trash Bluff Trash Push Finesse" that lets Bob, Donald play their slot 1, and Emily performs the trash push), but usually the double bluff interpretation blocks this, and all unnecessary moves usually get two cards (after Bob and Cathy play, Donald will stop and the one-away card still has to be gotten in some other way). Here it's clear that the double bluff interpretation is a bit of a compromise.


_Unnecessarily Using Known Trash_ (UB in the proposal) generalizes
* [*Unnecessary Trash Push*]
(https://github.com/Zamiell/hanabi-conventions/blob/master/Reference.md#the-unnecessary-trash-push)

Adds (without needing new definitions) moves that might be called
* *Unnecessary Trash Pull*
* *Unnecessary Trash Push Discharge*
* *Unnecessary Bad Chop Move Ejection*
* *Unnecessary Known Trash Ejection*
That all ask for a second blind play besides the original one. All these moves are originally 1-for-1, and relatively rarely used, so giving them clear 2-for-1 interpretations (for when they're unnecessarily used in the place of a direct clue) without adding complexity will make them more versatile.

**Comment by pianoblook** [comment]
Question about how 5ND could/should relate to these new Trash principles.

Based on my understanding of the history&logic of 5ND, the blindplay is called for in order to demonstrate a 'stop order' on the focused 5-Pull card. But this Discharge can occur on any >1AFP card, not just trash. If all other Discharges in the game directly correspond to _targeting Trash_, why have 5ND be an exception? Why, then, wouldn't a _Trash Push Discharge_ similarly revert the focused card to a 'trash or more than 1-away-from-playable'?

I suspect the reason is that the 5ND is inherently extra by getting a 5 clued (a ≥2 for 1), while every other 'necessary' Discharge is just a 1-for-1? They are still better than the alternative, though, if getting the 3rd-slot card would have otherwise been inefficient anyway.

TL;DR: Changing 5ND to being a response to 5-Pulling exactly _unknown trash_ (instead of any >1AWP) would reinforce the 'Trash Clue Principle'. Every Discharge would be unified as being: "If the _focus_ of the clue is trash, a Discharge is called for".

PRO: Super easy to remember and unifies all Discharge logic
PRO: any 5ND would result in a known trash card being marked as a bonus
CON: The chances of this actually arising would go way down. Like, probably never.

Lol as an added bonus, this could result in 5-Pull Ejections being back on the table in brown/null type suits.

**Comment by Zamiell** [comment]
piano 5ND is only turned on in the early game, and there is (almost) no trash in the early game, so im not sure that makes sense

**Comment by pianoblook** [comment]
Agreed, though I wonder if the logical rework/simplification could help spur on other clever advanced plays. 
e.g.:
Turn 1
Alice clues 2+ 1's to Donald
Bob clues an offchop 5 to Alice, 5-pulling a 1 that is duplicated
Cathy _Discharges_, and now the team knows that Alice's 1 is exactly a dupe

**Comment by Dr-Kakashi** [comment]
I've added some more notes.  Mainly just extra examples for clarification.

**Comment by Dr-Kakashi** [comment]
Kyle brought up some good points and I want to address it here:

![2020-06-01_2228](https://user-images.githubusercontent.com/48993588/83483543-bee4d800-a457-11ea-9254-e1036737a242.png)

In this example, you can see that Kyle can see it's "unnecessary."  Because of this Kyle will blind play.   However, if Kyle blind plays slot 1, then it undo's the trash push with what I've proposed.  What I've failed to realize is that after the first blind play the interpretation should be the known trash variant of the clue.  

What this essentially means is that the first blind played is ignored.  
- If Kyle, deems the push to be "unnecessary" as the image in this post shows, then Kyle should play slot 1.  
- If Kyle can see that the push card is trash, then he should treat this as the Unnecessary Trash Push Discharge and Discharge to "undo" the trash push.

**Comment by Dr-Kakashi** [comment]
Added Edit 2 to the original post:
These principles are mainly for early-mid game. Shall we make an exception where, once end game begins, only the trash double bluff conventions shall be used and unnecessary principles turns off?

**Comment by florian-5f3759df** [comment]
Fully converting to the known trash version is a bit more extreme, since the principles then imply things like a discharge + ejection (unnecessary UTD inducing a bad chop move ejection) or discharge + discharge (unnecessary UTD inducing a trash push discharge). It would make for a simple rule though, so this will not even be that confusing in game.

I would be happy with either the blind play cancellation compromise, as in the original proposal, and the version where we fully commit to the trash-clue interpretation.

**Comment by pianoblook** [comment]
I'm confused. First of all I guess it should be clarified whether kimbi has negative color info on her red cards. Otherwise, why wouldn't Lilliana immediately be prompted to _Discharge_ as an UTD? Otherwise if Lilliana plays slot 1 kimbi would mark it as a m2 bluff, no? And slot 2 would mark it as a potential m5 5CE.

Assuming it's already known non-multi, then let me see if I can write out what feels like should happen from the above:

1. Lilliana plays Slot 1 as a _Trash Bluff_. Kyle sees this whole line was unnecessary and therefore plays Slot 1. Despite this second blindplay, kimbi trusts this was not a double trash bluff and carries out the _Trash Push_.

2. Lilliana plays Slot 1 as a _Trash Bluff_. Kyle sees this was unnecessary (and suboptimal) and...misplays m3 from slot 2 as a _Push Ejection_ now that the red is _Known Trash_ and otherwise would trigger a _Trash Push_ (?)

3 & 4. Lilliana plays Slot 2 as a _Push Ejection_. Kyle sees this was unnecessary (and suboptimal) and plays slot 1 as a _Unnecessary Trash Move Finesse_

What am I missing here? Feel free to reach out via Discord if it's just me who's confused here haha.

**Comment by florian-5f3759df** [comment]
Your point about multi is correct, I presume Kakashi meant it as have negative multi. Your descriptions of 2,3,4 should not happen, since the pushed card is one away from playable. Therefore a trash push on the card should be interpreted as a trash push finesse, and not as an ejection.

**Comment by pianoblook** [comment]
Okay that would make a lot more sense, lol. I should have figured.

Still though, I don't think I'd agree that those red clues should count as 'unnecessary'. On the contrary, they very conveniently set up kimbi's b2 to be clued/finessed kimbi's.

Examples:
- In _Example 3_ Lilliana can now cleanly hint '2' to kimbi directly
- In _Example 4_ Kyle can now cleanly hint or bluff (e.g. black to Kakashi)

**Comment by asaelr** [comment]
In the last example in the original post, why does Cathy play slot 1? shouldn't she play her (new) chop?
Also, I disagree with the erasing of the 2nd bullet point, and the whole logic from there on. The created system prefer triple-plays on double-plays, and I think it should not be the case.
(Also, Alice should not be able no discharge Cathy)
What I suggest is that (I hope that I understood correctly the UA/UB terms...):
1. There should be no different between known trash and unknown trash that can't be one-away from playable (except that the latter situation is always UB)
2. In UA situations, the clue receiver should treat it as known trash (cm or push)
3. In UA situations, the next player (after the first blind-play) can blind play, to undo the normal known trash meaning.
4. unnecessary trash push finesse should either be canceled, or have an extra blind-play AFTER the trash push. (this is a special situation, when UA and UB mix)

**Comment by florian-5f3759df** [comment]
Maybe it's good to change the UA and UB names to something else, since it's hard to derive which is which from the names. The point for the split was that an unnecessary trash bluff is a different type of convention than an unnecessary trash push, and it's important to have a clear conceptual distinction. As an attempt, I'll use Unnecessarily using Unknown Trash (UUT) for UA, i.e., trash bluffs and UTD, and Unnecessarily using Known Trash (UKT) for UB, i.e., everything else.

For your question: In the proposal, the clue in the last example has known trash as focus. So it's UKT and asks for an extra slot 1 blind play, on the first appropriate person. Bob already has a blind play from the original trash push discharge (and Bob cannot know if it's unnecessary or not, since that depends on Bob's hand), so Cathy gets the bonus blind play.

You could of course say: "But the _real focus_ of Alice's clue was the trash pushed card, which has now turned into known trash by Bob's play. Shouldn't this be _actually_ case UUT?" That is a fair point that I kind of agree with, but I think it will make the convention much simpler if it's just defined from the focus of the original clue.

Of your interpretation, point 3 was indeed the proposal before the strikeout. Kakashi's proposed update changed that part and removed cancellations in favour of distinguishing between the blind play / ejection / discharge on Cathy. Like I said I'm fine with either the original version with the cancellation blind play, or the fully-committed version.

To point 4: A trash push finesse has trash as focus, so unnecessary trash push finesse is pure UKT. It just calls for three blind plays, it's not a mixture of UA and UB and there are no cancellations involved or possible. (The problem only happens if it's a unnecessary trash bluff on a chop card that's next to a one-away card. Then the original version's second blind play is identical for a cancellation or a trash push finesse. This is of course quite rare.)

**Comment by asaelr** [comment]
Thanks for the clarification, indeed I didn't understand it correctly earlier. However, I don't think that this is the correct conceptual distinction.
As far as I can see, we try to develop a system where a trash clue can have 2 (or even 3) implications. For example, it can imply a blind play from the next player, and (in additional) a chop move from the receiver of the clue. So, in order to speak about that, we must consider for each of them if it could have been achieved using a normal (non-trash) clue. If so, there is an unnecessity in respect to that implication.
(This, in fact, the main difference between unnecessary trash bluff and unnecessary trash push - in UTB the first implication could have been achieved normally, and in UTP it's the second implication)

We can also define different convention based on the knownness of the clue target (to the receiver), but I don't think we should. If Alice clued Donald a trash card, and Bob played, Donald now knows it's a trash. Why should Cathy and Donald change their actions based on whether Donald knew it's a trash before of the blind-play? I think that such system is much more complex than needed.
(I would suggest terms for these two kind of unnecessity, but we already have too much terms...)

About the example: I still am not sure I understand. Should Cathy blind-play slot 1 _in addition_ to the push? (both You and Kakashi (in the original post) didn't said so explicitly, but in the comment with the 4 cases and pictures, it's clearly the situation. Also, if not, why call that 'push'? So I gonna assume yes here)
Which card should she play first?
I think that it will be too problematic for a trash-push to not play immediately, but in that case, why should Cathy do the extra blind-play? I would think it's on Donald (or 2nd finesse position of Bob, in 3p game).
But that's also related to my (probably) main point: we shouldn't develop a system that prefer triple-plays on double-plays. Triple-play situation are much more rare, and we shouldn't define a convention that achieve 3 blind-plays, if it doesn't have any simpler convention, that gets 2 of them.
(i.e. that example practically kills the unnecessary trash push as we know it)

I agree that the unnecessary trash push finesse is not really a mixture of the unnecessity types. it's just a contradiction of the cancellation with the second implication.
So the options are: either cancel that convention (possible, but I don't like it), change the cancellation to be a discharge instead of finesse position (possible, but it kills the trash double bluff convention, and I'm totally against it), or to delay the extra blind-play until after the trash-push (my preferred solution).
Anyway, this point isn't related to the UKT/UUT thing, and we can make different decisions in these two topics.

**Comment by florian-5f3759df** [comment]
I think there is a miscommunication about the example still.

For the example, Alice clues a trash card on Cathy's chop (say slot 4), for example a 1 after all 1s are played. This would be a trash push. Now Bob plays slot 3 to show the card that Cathy's card she's about to play (in slot 3) is trash, so clearly Cathy shouldn't be playing it, and has no blind play so far. 

Now Cathy notices that Bob's slot 3 could have easily been clued directly, instead of using this weird trash push discharge. So Cathy performs a blind play of slot 1. (Not slot 2, and especially not slot 3, since Bob's discharge just told Cathy she has trash on slot 3.)

So there is no 'in addition' to the push, and Cathy has a single blind play, not two. And the team plays two cards in total, a discharge on Bob and finesse position of Cathy.

**Comment by asaelr** [comment]
Thanks. I think I understand it now.

Another point I just thought of is that we need to define what exactly is considered "unnecessary".
What if there is a direct play clue, that touch some extra trash card? (my proposal - necessary, unless the trash would be known)
What if I can give an ambiguous play clue that will take forever to resolve? (I don't have any proposal here. It's even more complicated if someone doesn't know it would be ambiguous)
What if there is a direct play clue that touch an extra duplicated card? (probably we should treat it similar to the good touch bluff - if I trash bluff b3 instead of cluing blue to b3+b4, someone is promised to have a clued b4)

**Comment by Zamiell** [comment]
can either kakashi or florian write a doc entry for this

**Comment by Dr-Kakashi** [comment]
I will talk with florian and we'll work one up.

**Comment by Dr-Kakashi** [comment]
> Your point about multi is correct, I presume Kakashi meant it as having negative rainbow. Your descriptions of 2,3,4 should not happen, since the pushed card is one away from playable. Therefore a trash push on the card should be interpreted as a trash push finesse, and not as an ejection.
> 
> Edit: Oh wait, apologies. Only the first picture has g1 already played. I think going by the description kakashi just forgot to include g1 (I didn't even see it, since it was correct in the first picture)...

I've fixed the pictures

**Comment by Dr-Kakashi** [comment]
> Another point I just thought of is that we need to define what exactly is considered "unnecessary".
It's unnecessary if a normal play clue can be given or normal save could be given. 

> What if there is a direct play clue, that touch some extra trash card? (my proposal - necessary, unless the trash would be known)
Then it isn't unnecessary and doesn't apply here.  These trash principles pertain to situations where a trash convention is used that are normally 1 for 1's.

> What if I can give an ambiguous play clue that will take forever to resolve? (I don't have any proposal here. It's even more complicated if someone doesn't know it would be ambiguous)
I don't know how an ambiguous situation could occur as the next person must immediately blind play.  

> What if there is a direct play clue that touch an extra duplicated card? (probably we should treat it similar to the good touch bluff - if I trash bluff b3 instead of cluing blue to b3+b4, someone is promised to have a clued b4)
Then it's unnecessary and falls under these principles.  Whether or not b4 elimination notes should be written...I'm unsure.  I want to say no because these unnecessary principles are at least a 2 for 1 clue, which would be the same amount, if the b3 & b4 were just clued directly.

**Comment by padiwik** [comment]
If Alice clues known trash to Donald's hand, chop moving trash, and it's unnecessary. Does this call for Bob and Cathy to play slot 1, and it's resolved? Because it can't be like a trash push finesse anymore

**Comment by florian-5f3759df** [comment]
The initial interpretation always stays the way it's defined now, so it's a bad chop move ejection. Then Cathy can look at Bob's hand, and see that Alice could have just clued the card directly, and therefore plays slot 1 as a bonus blind play.

Note that Bob has no idea whether his slot 2 is cluable or not, so you can't expect Bob to base his blind play on whether it's necessary or not. So it doesn't make sense for Bob to play slot 1 instead of the usual slot 2.

**Comment by pianoblook** [comment]
<img width="1280" alt="Screen Shot 2020-06-22 at 2 35 50 AM" src="https://user-images.githubusercontent.com/32303453/85255910-31325200-b431-11ea-8d7a-29e1f809794e.png">

As if this needs any more evidence....

**Comment by Zamiell** [comment]
its been 25 days since i requested a pr, i will leave this open for another 30 days, if no pr by them i will close this issue and we will consider it denied

**Comment by Dr-Kakashi** [comment]
**Unnecessary Convention Principles**

- The following section requires players to have full understanding of trash conventions.  Trash Bluff/Finesse, UTD, Trash Push, Trash Chop Move, etc.
- Trash conventions are usually done in situations where cards are hard to be play clued directly:
- Blocked by other cards or even their own suit (brown or null)
- Blocked by cards in the trash pile such that a play clue looks like a save clue
- Generally, when a trash convention is used they are 1 for 1 and may or may not give any useful information to the clue receiver.
- Thus, any convention that is a 1 for 1 and does not directly clued a card also falls under this category.  Ambiguous blue clue for example.
- We also can agree that Trash Conventions are more complicated/advanced compared to a normal play/save clue. 
- If a trash convention is used when a normal clue would suffice, we may call this situation “unnecessary.” 

Definition of Unnecessary: 

1. A regular play/save clue could’ve been given to the blind played card
2. No other useful cards are picked up 
3. No extra information is given to the clue receiver’s hand 
   - Such as fixing an impending misplay
   - Adding positive information to their hand

If **known trash** was used, the first player who can see that the clue was unnecessary should blind play slot 1. (This will be the first player that does not already have a blind play from the original clue.)

- Should this trash chop move or trash push be wrong, this player can play slot 3 as a discharge, since they are the first to act.
- If there is an in-between player and a trash push was initiated, then there are two situations
  - The two cards are unrelated - the in-between player does nothing. (Unnecessary Trash Push)
  - The two cards are related - this is unnecessary from their point of view and they should blind play slot 1, which will not affect the information in the clue receiver’s hand, meaning the clue receiver will still trash push.

If **unknown trash** was used (a Trash Bluff or Unnecessary Trash Discharge), we agree that in addition to the blind play from the first player, the second player, who received the clue, should treat it as a trash chop move or trash push depending on the location of the trash in the hand. 

- The first blind play is necessary to tell the clue receiver that their focused card is trash.  This means they have no way to tell them to not cm or trash push.  Thus, the in-between player needs to have a way to cancel out the move.
- If there is an in-between player, then there are two situations:
  - If the chop moved cards are useful or the trash pushed card is playable, then they should do nothing.  
  - If the chop moved cards are trash or the trash pushed card is trash, they should immediately blind play slot 1, to immediately resolve it as a trash double bluff.  This is necessary in order to undo the trash convention.

**Examples:**
Unnecessary from Scharkbite’s Perspective using of Trash Push: an Unnecessary Trash Push
![2020-06-15_1141](https://user-images.githubusercontent.com/48993588/87857040-6e76ec00-c8d8-11ea-89fd-94d71bef6896.png)

Unnecessary from Floriman’s Perspective using the Bad Chop Move Ejection
![2020-06-15_1134](https://user-images.githubusercontent.com/48993588/87857043-733ba000-c8d8-11ea-8ee8-8e9bc82bcb93.png)

**Comment by Zamiell** [comment]
i need a pull request, not just copy pasting what is in the op
